token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -800,13 +806,269 @@ fn process_allow_balance_credits(
+    let instructions_sysvar_info = next_account_info(account_info_iter)?;
+    let authority_info = next_account_info(account_info_iter)?;
+    let authority_info_data_len = authority_info.data_len();
+    let account_infos = account_info_iter.as_slice();
+    let num_signers = account_infos
+        .len()
+        .saturating_sub(num_token_accounts as usize);
+
+    let mut mint_data = mint_account_info.data.borrow_mut();
+    let mut mint = StateWithExtensionsMut::<Mint>::unpack(&mut mint_data)?;
+
+    // mint must be extended for fees
+    let fee_mint = mint.get_extension::<TransferFeeConfig>()?;
+    let withdraw_withheld_authority = Option::<Pubkey>::from(fee_mint.withdraw_withheld_authority)
+        .ok_or(TokenError::NoAuthorityExists)?;
+    Processor::validate_owner(
+        program_id,
+        &withdraw_withheld_authority,
+        authority_info,
+        authority_info_data_len,
+        &account_infos[..num_signers],
+    )?;
+
+    let mut dest_account_data = dest_account_info.data.borrow_mut();
+    let mut dest_account = StateWithExtensionsMut::<Account>::unpack(&mut dest_account_data)?;
+    if dest_account.base.mint != *mint_account_info.key {
+        return Err(TokenError::MintMismatch.into());
+    }
+    if dest_account.base.is_frozen() {
+        return Err(TokenError::AccountFrozen.into());
+    }
+
+    // sum up the withheld amounts in all the accounts
+    let mut aggregate_withheld_amount = EncryptedWithheldAmount::zeroed();
+    for account_info in &account_infos[num_signers..] {
+        // self-harvest, can't double-borrow the underlying data
+        if account_info.key == dest_account_info.key {
+            dest_account
+                .get_extension::<TransferFeeAmount>()
+                .map_err(|_| TokenError::InvalidState)?;
+
+            let confidential_transfer_dest_account = dest_account
+                .get_extension_mut::<ConfidentialTransferAccount>()
+                .map_err(|_| TokenError::InvalidState)?;
+
+            aggregate_withheld_amount = ops::add(
+                &aggregate_withheld_amount,
+                &confidential_transfer_dest_account.withheld_amount,
+            )
+            .ok_or(ProgramError::InvalidInstructionData)?;
+
+            confidential_transfer_dest_account.withheld_amount = EncryptedWithheldAmount::zeroed();
+        } else {
+            match harvest_from_account(mint_account_info.key, account_info) {
+                Ok(encrypted_withheld_amount) => {
+                    aggregate_withheld_amount =
+                        ops::add(&aggregate_withheld_amount, &encrypted_withheld_amount)
+                            .ok_or(ProgramError::InvalidInstructionData)?;
+                }
+                Err(e) => {
+                    msg!("Error harvesting from {}: {}", account_info.key, e);
+                }
+            }
+        }
+    }
+
+    let mut confidential_transfer_dest_account =
```

Comment:
Nit: This variable was named differently in the preceding function:
```
let mut dest_confidential_transfer_account =
        dest_account.get_extension_mut::<ConfidentialTransferAccount>()?;
```
Please use one or the other

Reply:
Fixed.